android/cli/commands/_build.js

```diff
@@ -1472,24 +1472,32 @@ AndroidBuilder.prototype.validate = function validate(logger, config, cli) {
 				if (module.platform.indexOf('commonjs') != -1) {
 					module.native = false;
 
-					// Look for legacy module.id.js first
-					module.libFile = path.join(module.modulePath, module.id + '.js');
-					if (!fs.existsSync(module.libFile)) {
-						// then package.json TODO Verify the main property points at reale file under the module!
-						module.libFile = path.join(module.modulePath, 'package.json');
-						if (!fs.existsSync(module.libFile)) {
-							// then index.js
-							module.libFile = path.join(module.modulePath, 'index.js');
-							if (!fs.existsSync(module.libFile)) {
-								// then index.json
-								module.libFile = path.join(module.modulePath, 'index.json');
-								if (!fs.existsSync(module.libFile)) {
-									this.logger.error(__('Module %s version %s is missing module files: %s, package.json, index.js, or index.json', module.id.cyan, (module.manifest.version || 'latest').cyan, path.join(module.modulePath, module.id + '.js').cyan) + '\n');
-									process.exit(1);
+					// look for legacy module.id.js first
+					var libFile = path.join(module.modulePath, module.id + '.js');
+					module.libFile = fs.existsSync(libFile) ? libFile : null;
+					if (!module.libFile) {
+						var pkgJsonFile = path.join(module.modulePath, 'package.json');
+						if (fs.existsSync(pkgJsonFile)) {
+							try {
+								var pkgJson = require(pkgJsonFile);
+								if (pkgJson && pkgJson.main && fs.existsSync(libFile = path.join(module.modulePath, pkgJson.main + (/\.js$/.test(pkgJson.main) ? '' : '.js')))) {
```

Comment:
So this clearly is doing more validation that the package.json has a main property and checking to see if the main property file exists, which we'd want.

Because package.json is used by require logic, we can figure out the possible valid values for main property sort of indirectly here: https://nodejs.org/api/modules.html#modules_all_together

Effectively we know it tries to load the value as a file, which means the test for no extension and then appending '.js' wouldn't be right. We need to try, in order:
```js
path.join(module.modulePath, pkgJson.main);
path.join(module.modulePath, pkgJson.main + '.js');
path.join(module.modulePath, pkgJson.main + '.json');
```